src/librustdoc/html/markdown.rs

```diff
@@ -1123,7 +1123,13 @@ crate fn plain_text_summary(md: &str) -> String {
     s
 }
 
-crate fn markdown_links(md: &str) -> Vec<(String, Range<usize>)> {
+pub struct MarkdownLink {
+    pub kind: LinkType,
+    pub link: String,
+    pub range: Range<usize>,
+}
```

Comment:
Please make this crate-private so we get warnings if it's unused.
```suggestion
crate struct MarkdownLink {
    pub kind: LinkType,
    pub link: String,
    pub range: Range<usize>,
}
```